whois-nrtm4/src/main/java/net/ripe/db/nrtm4/SnapshotFileGenerator.java

```diff
@@ -45,51 +48,56 @@ public SnapshotFileGenerator(
         this.nrtmFileStore = nrtmFileStore;
     }
 
-    public Optional<PublishableSnapshotFile> createSnapshot(final NrtmSource source) throws IOException {
+    public List<PublishableSnapshotFile> createSnapshots() throws IOException {
         // Get last version from database.
-        final Optional<NrtmVersionInfo> lastVersion = nrtmVersionInfoRepository.findLastVersion(source);
-        NrtmVersionInfo version;
+        final Optional<NrtmVersionInfo> lastVersion = nrtmVersionInfoRepository.findLastVersion();
+        final List<NrtmVersionInfo> sourceVersions = new ArrayList<>();
         LOGGER.info("lastVersion.isEmpty() {}", lastVersion.isEmpty());
         if (lastVersion.isEmpty()) {
-            version = snapshotObjectSynchronizer.initializeSnapshotObjects(source);
-            nrtmFileStore.createNrtmSessionDirectory(version.getSessionID());
-        } else {
-            version = lastVersion.get();
-            if (version.getType() == NrtmDocumentType.DELTA) {
-                LOGGER.info("Not generating snapshot file yet");
-                return Optional.empty();
-            } else {
-                LOGGER.info("Not generating snapshot file since no deltas have been published since v{} with serialID {}",
-                    version.getVersion(), version.getLastSerialId());
-                return Optional.empty();
+            final InitialSnapshotState state = snapshotObjectSynchronizer.initializeSnapshotObjects();
+            for (final NrtmSource source : NrtmSourceHolder.getAllSources()) {
+                final NrtmVersionInfo version = nrtmVersionInfoRepository.createInitialVersion(source, state.serialId());
+                nrtmFileStore.createNrtmSessionDirectory(version.getSessionID());
```

Comment:
is this needs o be inside if condition?  We always have to create a version even if its not initial snapshot? So this logic remains same as for publishable file generation?

Reply:
Latest commit uses only NrtmSourceModel to represent a source and all functions are in SourceRepository. Don't worry about how the code in this section is now -- it will change a lot in the next story when we won't have a snapshot_object table (so initializeSnapshotObjects will be removed)